application/src/main/data/upgrade/basic/schema_update.sql

```diff
@@ -209,3 +209,75 @@ $$;
 UPDATE resource SET resource_sub_type = 'EXTENSION' WHERE resource_type = 'JS_MODULE' AND resource_sub_type IS NULL;
 
 -- UPDATE RESOURCE JS_MODULE SUB TYPE END
+
+-- UPDATE SAVE TIME SERIES NODES START
+
+DO $$
+    BEGIN
+        -- Check if the rule_node table exists
+        IF EXISTS (
+            SELECT 1
+            FROM information_schema.tables
+            WHERE table_name = 'rule_node'
+        ) THEN
+
+            -- CREATE JSON validation function
+            CREATE OR REPLACE FUNCTION is_valid_jsonb(input text)
+                RETURNS boolean
+                LANGUAGE plpgsql
+            AS $func$
+            DECLARE
+                dummy JSONB;
+            BEGIN
+                dummy := input::jsonb;
+                RETURN true;
+            EXCEPTION
+                WHEN others THEN
+                    RETURN false;
+            END;
+            $func$;
+
+            UPDATE rule_node
+            SET configuration = CASE
```

Comment:
This is a bit over-complicated. I suggest to separate this one "super-statement" to multiple statements with clear "WHERE" conditions. This will simplify the script, readability and execution. No need to support case 2 (how this can happen?).

Reply:
How the JSON may be invalid? Why do we need to support such case?